src/algorithms/ctmrg.jl

```diff
@@ -1,24 +1,46 @@
+# TODO: add option for different projector styles (half-infinite, full-infinite, etc.)
+"""
+    struct ProjectorAlg{S}(; svd_alg = TensorKit.SVD(), trscheme = TensorKit.notrunc(),
+                           fixedspace = false, verbosity = 0)
+
+Algorithm struct collecting all projector related parameters. The truncation scheme has to be
+a `TensorKit.TruncationScheme`, and some SVD algorithms might have further restrictions on what
+kind of truncation scheme can be used. If `fixedspace` is true, the truncation scheme is set to
+`truncspace(V)` where `V` is the environment bond space, adjusted to the corresponding
+environment direction/unit cell entry.
+"""
+@kwdef struct ProjectorAlg{S}
+    svd_alg::S = TensorKit.SVD()
+    trscheme::TruncationScheme = TensorKit.notrunc()
+    fixedspace::Bool = false
+    verbosity::Int = 0
+end
```

Comment:
```suggestion
@kwdef struct ProjectorAlg{S,T}
    svd_alg::S = TensorKit.SVD()
    trscheme::T = TensorKit.notrunc()
    verbosity::Int = 0
end
struct FixedSpaceTruncation <: TensorKit.TruncationScheme end
```
How about something like this, where we absorb the "fixedspace" option into the `trscheme`? In any case, I don't think it makes sense to have both `fixedspace` and a non-trivial `trscheme`.
As a side note, I would assume that the default option of `fixedspace=true` makes more sense than `notrunc`, considering the fact that `notrunc` would necessarily run out of memory at some point.